### PR TITLE
kvclient: fix data race about checkpoint ts

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
fix data race about checkpoint ts

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -543,6 +537,15 @@ func (c *CDCClient) singleEventFeed(
 	}
 }
 
+func updateCheckpointTS(checkpointTs *uint64, newValue uint64) {
+	for {
```
Comment (amyangfei) [inline cdc/kv/client.go]: no need to use the for loop?

Comment (leoppro) [reply cdc/kv/client.go]: yes, we need

we need retry if `CompareAndSwap` is failed.

Comment (amyangfei) [reply cdc/kv/client.go]: right, found the `checkpointTs` can be updated by both background sorter and when receiving resolvedTs

**Pull request merged**